Source/WebCore/page/scrolling/AsyncScrollingCoordinator.cpp

```diff
@@ -516,6 +513,21 @@ LocalFrameView* AsyncScrollingCoordinator::frameViewForScrollingNode(ScrollingNo
     }
 
     return nullptr;
+
+}
+
+LocalFrameView* AsyncScrollingCoordinator::frameViewForScrollingNode(ScrollingNodeID scrollingNodeID) const
+{
+    if (!m_scrollingStateTree->rootStateNode())
+        return nullptr;
+
+    for (const WeakRef<LocalFrame>& rootFrameRef : page()->rootFrames()) {
+        if (auto* rootFrame = rootFrameRef.ptr()) {
+            if (auto* frameView = frameViewForScrollingNode(*rootFrame, scrollingNodeID))
```

Comment:
if (!page())
    return nullptr;
for (const auto& rootFrame : page()->rootFrames()) {
    if (RefPtr frameView = frameViewForScrollingNode(rootFrame.get(), scrollingNodeID))